### ISSUE TITLE
Socket activation with systemd

### ISSUE THREAD

**Issue by imilos** [kravietz/poppassd-ceti#7]
Dear Pawel,

First to thank you for doing maintenance of this old but useful stuff! Have you considered promoting systemd socket activation instead of doing it "old way" using xinetd/initd? I wrote a socket file:

```
# /etc/systemd/system/poppassd.socket
[Unit]
Description=Poppassd Socket

[Socket]
ListenStream=0.0.0.0:106
Accept=yes

[Install]
WantedBy=sockets.target
```
and a service file as well:
```
# /etc/systemd/system/poppassd@.service
[Unit]
Description=Poppassd Service
Requires=poppassd.socket

[Service]
Type=simple
ExecStart=/usr/local/sbin/poppassd
StandardInput=socket
StandardOutput=socket
StandardError=journal
TimeoutStopSec=5
#RuntimeMaxSec=10

[Install]
WantedBy=multi-user.target
```
The setup is working well on a production server. 

Kind regards
Milos

**Comment by kravietz** [comment]
@imilos Thanks a lot, yes absolutely - `xinetd` was the top innovation back when the project was started but it certainly deserves an update :+1:

**Comment by kravietz** [comment]
@imilos Can you possibly try the hardened `systemd` service file [poppassd.service](https://github.com/kravietz/poppassd-ceti/blob/master/etc/systemd/poppassd.service)? It works for me but things work differently on different systems...

**Comment by imilos** [comment]
I've just tried it on my Ubuntu 20.04 system. However: 

```
[milos@mmilos debian]$ telnet localhost 106
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
200 poppassd
user milos
200 Your password please
pass *****
200 Your new password please
newpass ********
500 Server pam_chauthtok error 20, password not changed
```
I do not have much experience with service hardening...

**Comment by kravietz** [comment]
@imilos Try the same while `journalctl -f`, all PAM errors are now sent to logs instead of returning to the client.

**Comment by imilos** [comment]
For some reason I had to rename your `popassd.service` into `poppassd@.service` to get it working. However, nothing more informative in the journal:
```
авг 31 23:32:30 mmilos systemd[1]: poppassd@3-127.0.0.1:106-127.0.0.1:60526.service: Main process exited, code=exited, status=1/FAILURE
авг 31 23:32:30 mmilos systemd[1]: poppassd@3-127.0.0.1:106-127.0.0.1:60526.service: Failed with result 'exit-code'.
```

**Comment by kravietz** [comment]
@imilos socket file naming seems to be required by [systemd.socket](https://www.freedesktop.org/software/systemd/man/systemd.socket.html), it took me a while to figure out too:

> Example: a socket file foo.socket needs a matching service foo.service if Accept=no is set. If Accept=yes is set, a service template foo@.service must exist from which services are instantiated for each incoming connection.

Did you install the new files using `cmake` or just replaced the service file?

**Comment by imilos** [comment]
@kravietz thank you, yes, it's a template thing, with `@` automatically replaced with the number. I did a clean install using `master` branch and `cmake`. I only added `@` to `poppassd@.service`, as I already stated. If I remove all the security statements you've added, it starts working. 

Furthermore, I tried the same on Fedora 32 (cmake build). Firstly, it does not recognize PAM settings on Fedora:
```
Sep 01 08:44:37 radojica poppassd[15157]: PAM (poppassd) illegal module type: @include
Sep 01 08:44:37 radojica poppassd[15157]: PAM pam_parse: expecting return value; [...common-password]
Sep 01 08:44:37 radojica poppassd[15157]: PAM (poppassd) no module name supplied

```
ending with `500 Old password is incorrect`. But, If I put:
```
auth       include	system-auth
account    include	system-auth
password   include	system-auth
```
into `/etc/pam.d/poppassd` and remove all security statements, it starts working. 

I hope this helps, at least a bit.

**Comment by kravietz** [comment]
@imilos Ok I tested it now and it seems Ubuntu 20 uses this `@include` syntax but my Manjaro uses `password include` syntax... But the `systemd` security statements still fail it for you?

**Comment by anq4** [comment]
both the here proposed /etc/systemd/system/poppassd@.service and the included /etc/systemd/system/poppassd.service just do not work on rhel/centos 8

the former does nothing and the latter gives a socket error

systemd[1]: poppassd.service: Got no socket.

**Comment by bmanojlovic** [comment]
I have fixed service file in #14 and i just replace pam file with my own (same as what @imilos was using)

**Comment by kravietz** [comment]
Thank you @bmanojlovic merged your change!
